ksqldb-engine/src/main/java/io/confluent/ksql/analyzer/AggregateAnalyzer.java

```diff
@@ -276,17 +273,31 @@ private void enforceAggregateRules() {
     }
   }
 
+  /**
+   * This visitor performs two tasks: Create the input schema to the AggregateNode and validations.
+   *
+   * For creating the input schema, it checks if any expression along the path from the root
+   * expression to the leaf (UnqualifiedColumnReference) is part of the groupBy. If at least one is,
+   * then the UnqualifiedColumnReference is added to the schema.
+   *
+   * For validation, the visitor checks that:
+   * 1) expressions in non-aggregate functions are part of the grouping clause,
```

Comment:
```suggestion
   * 1) expressions not in aggregate functions are part of the grouping clause,
```

Reply:
It's actually correct what I wrote :P  It will throw if expressions used in aggregate functions are not part of the group by

Reply:
>  It will throw if expressions used in aggregate functions are not part of the group by

now I'm more confused 😂 expressions in aggregate functions must not be part of the group by - `SELECT id, SUM(foo) FROM s GROUP BY id` works (foo is an aggregate function parameter not in group by)

What you had originally written is true, but does not cover everything. It says expressions in UDFs (not UDAFs) must be part of the grouping clause (true). But it is also true that even if they're not in a UDF, as long as they're not in an aggregate function, they must be in the grouping clause:

```sql
--- valid
SELECT non_agg_fun(foo), COUNT(*) from s GROUP BY foo;
SELECT foo, COUNT(*) from s GROUP BY foo;

-- invalid
SELECT foo, COUN(T*) from s GROUP BY bar;
```

Reply:
Ah sorry, I meant expressions in non-aggregate functions must be part of group-by. So much confusion :) So, the comments say that the validator will throw if any of the conditions does not hold

Reply:
Ah wait, I think I know get what you are suggesting. You are suggesting to make the sentence more general. OK, will make the change